scripts/imgtool/image.py

```diff
@@ -0,0 +1,164 @@
+"""
+Image signing and management.
+"""
+
+from __future__ import print_function
+
+import hashlib
+import struct
+
+IMAGE_MAGIC = 0x96f3b83c
+IMAGE_HEADER_SIZE = 32
+
+# Image header flags.
+IMAGE_F = {
+        'PIC':                   0x0000001,
+        'SHA256':                0x0000002,
+        'PKCS15_RSA2048_SHA256': 0x0000004,
+        'ECDSA224_SHA256':       0x0000008,
+        'NON_BOOTABLE':          0x0000010,
+        'ECDSA256_SHA256':       0x0000020, }
+
+TLV_VALUES = {
+        'SHA256': 1,
+        'RSA2048': 2,
+        'ECDSA224': 3,
+        'ECDSA256': 4, }
+
+# Sizes of the image trailer, depending on image alignment.
+trailer_sizes = {
+        1: 402,
+        2: 788,
+        4: 1560,
+        8: 3104, }
+
+boot_magic = bytes([
+    0x77, 0xc2, 0x95, 0xf3,
+    0x60, 0xd2, 0xef, 0x7f,
+    0x35, 0x52, 0x50, 0x0f,
+    0x2c, 0xb6, 0x79, 0x80, ])
+
+class TLV():
+    def __init__(self):
+        self.buf = bytearray()
+
+    def add(self, kind, payload):
+        """Add a TLV record.  Kind should be a string found in TLV_VALUES above."""
```

Comment:
It will raise a KeyError. This does indicate a programming error, and doesn't come from the user, so I don't see a strong reason to generate a different exception.